SensusService/DataStores/DataStore.cs

```diff
@@ -282,6 +284,30 @@ public Task AddAsync(Datum datum, CancellationToken cancellationToken)
                     });
                 }
 
+                // if the probe's RunLocalCommitOnStore option is set to true, commit it locally
+                else if (runLocalCommitOnStore && !_sampleTriggeredCommitRunning)
+                {
+                    SensusServiceHelper.Get().Logger.Log("Running sample-triggered commit.", LoggingLevel.Normal, GetType());
+
+                    _sampleTriggeredCommitRunning = true;
+
+                    return Task.Run(async () =>
+                    {
+                        try
+                        {
+                            await CommitChunksAsync(_data, this, cancellationToken);
+                        }
+                        catch (Exception ex)
+                        {
+                            SensusServiceHelper.Get().Logger.Log("Failed to run sample-triggered commit:  " + ex.Message, LoggingLevel.Normal, GetType());
+                        }
+                        finally
+                        {
+                            _sampleTriggeredCommitRunning = false;
+                        }
+                    });
+                }
+
```

Comment:
Too much duplication of earlier code...might put the `Task.Run` into an `Action` that gets used in both places.